coordinator/provider/coordinator.go

```diff
@@ -256,22 +256,39 @@ func (qc *qdbCoordinator) watchRouters(ctx context.Context) {
 	}
 }
 
-// NewCoordinator side efferc: runs async goroutine that checks
-// spqr router`s availability
 func NewCoordinator(db qdb.XQDB) *qdbCoordinator {
-	cc := &qdbCoordinator{
+	return &qdbCoordinator{
 		db: db,
 	}
+}
+
+// RunCoordinator side efferc: runs async goroutine that checks
+// spqr router`s availability
```

Comment:
// RunCoordinator side effect: it runs an asynchronous goroutine 
// that checks the availability of the SPQR router